src/modules/routing/utils/getPath.js

```diff
@@ -37,7 +37,9 @@ const getPath = (to, routes) => {
 	const { params = {}, query = {} } = to;
 	let queryStringParts = [];
 	each(params, (paramValue, paramKey) => {
-		const replaced = result.replace(new RegExp(`:${paramKey}(?=\/|$)`), paramValue);
+		const replaced = !paramValue ?
+			result.replace(new RegExp(`/:${paramKey}\\?(?=\/|$)`), '') :
+			result.replace(new RegExp(`:${paramKey}\\??(?=\/|$)`), paramValue);
```

Comment:
pridano `\\??`. Nevim proc musi byt 2 zpetna lomitka a u lookahead to treba neni ale budiz.